src/jshint.js

```diff
@@ -2138,7 +2138,31 @@ var JSHINT = (function() {
   bitwise(">>", "shiftright", 120);
   bitwise(">>>", "shiftrightunsigned", 120);
   infix("in", "in", 120);
-  infix("instanceof", "instanceof", 120);
+  infix("instanceof", function(left, token) {
+    var right;
+    var scope = state.funct["(scope)"];
+    var redefinedUndefined = scope.has("undefined");
```

Comment:
Considering how rarely this value will matter, this operation is a little
expensive. I think `scope.has("undefined")` is readable enough to warrant
invoking it in-line (without a named binding like `redefinedUndefined`) from
the conditional.